add combination of vector and raster layer in lesson 4 (#141)
carpentries-incubator/r-geospatial-urban: episodes/16-basic-gis-with-r-sf.Rmd

Reviewer: I see some indentation issues throughout this episode. This is minor and can be addressed in a separate PR, so this one remains readable.

Reviewer: ```suggestion
bb <- getbb("Brielle, NL")
```

Reviewer: ```suggestion
sf_use_s2(FALSE)
```

```

## Combining vector and raster data

Another criterion to characterise the potential conservation zones would be their mean elevation level (if we tried to prioritise polder structures for instance).

> To do that, we need to use elevation values of the dtm (digital terrain model) we used in the rater lesson, select the cells corresponding to the different zone and compute their mean value.

Reviewer: ```suggestion
To do that, we need to use the elevation values of the DTM (digital terrain model) we used in the raster lesson, select the cells corresponding to the different zones, and compute their mean value.
```

If so happens that Brielle is located at the intersection of two tiles. Let's open and combine them.

```{r}

> library(terra)

Reviewer: This should be moved to the beginning of the episode to keep all `library()` calls in one place.

> dtm_brielle <- rast("data/Brielle_dtm.tif")

Reviewer: Make sure you use `here()`. This was one of the issues raised during beta testing (#130) and fixed in #142

> dtm_brielle <- rast("data/Brielle_dtm.tif")

Reviewer 2: Here we import only one tile. Do we combine them beforehand or do we ask the participants to merge them?

crs(dtm_brielle)

> dtm_brielle_df <- as.data.frame(dtm_brielle, xy = TRUE)

Reviewer 2: do we need a df here?

Reviewer: ```suggestion
```
Indeed, this does not seem to be used anywhere.

> dtm_brielle <- project(dtm_brielle,
>                                crs(single_buffer))

Reviewer: ```suggestion
dtm_brielle <- project(dtm_brielle, crs(single_buffer))
```

conservation_elevation <- extract(x = dtm_brielle, y = single_buffer, raw = FALSE)

summary(conservation_elevation)

ggplot() +

>   geom_histogram(data = conservation_elevation, aes(x = Brielle_dtm)) +

Reviewer 2: ```suggestion
  geom_histogram(data = conservation_elevation, aes(x = dtm_brielle)) +
```

Reviewer: @dcannatella, it is a bit confusing, but that is the name of the variable `conservation_elevation$Brielle_dtm`, not the data frame. @ClementineCttn maybe it helps to make it clear here what the `terra::extract()` function does in line 240, or to rename the resulting columns to avoid confusion.

  ggtitle("Histogram of elevation values in conservation zones (m)") +
  xlab("Elevation") +
  ylab("Frequency of Pixels")

avg_conservation_elevation <- extract(x = dtm_brielle, y = single_buffer,
                                fun = mean)

> 

Reviewer: ```suggestion
```
